dockers/docker-fpm-frr/start.sh

```diff
@@ -31,22 +13,7 @@ echo "# Config files managed by sonic-config-engine" > /var/sonic/config_status
 
 rm -f /var/run/rsyslogd.pid
 
-supervisorctl start rsyslogd
```

Comment:
This was puropsefully added to the start.sh script to ensure rsyslogd started *after* we remove the PID file. How is this ordering now enforced, or is it?

If it isn't, we could now add a one-shot process to supervisor.conf which deletes the PID file and then make rsyslod dependent upon that process completing.

Reply:
it is enforce here. https://github.com/Azure/sonic-buildimage/pull/4599/files#diff-38a9e937db6463fd421715b886271d71R39

Reply:
I see. In this case, do we even need a start.sh script anymore (at least for this container)? Can we not just move the remaining contents of start.sh into docker_init.sh?

Reply:
It make sense to merge start.sh and docker_init.sh. And start.sh is a better filename.

---
In reply to: [425468397](https://github.com/Azure/sonic-buildimage/pull/4599#discussion_r425468397) [](ancestors = 425468397)

Reply:
The idea is that `docker_init.sh` runs before supervisord starts, and prepares the container and then exec's supervisord. In retrospect, I think a better name for it is `container_init.sh`. Whereas `start.sh` was designed to be started by supervisor and it contains the logic to start the processes in order.

For this container, it looks like the only lines left in start.sh which can't go in `docker_init.sh` (or `container_init.sh` are:

```
# start eoiu pulling, only if configured so
if [[ $(sonic-cfggen -d -v 'WARM_RESTART.bgp.bgp_eoiu') == 'true' ]]; then
    supervisorctl start bgp_eoiu_marker
fi
```

Instead we could put these in their own shell script and have supervisor start it as a process.

Reply:
anyway, I moved rsyslogd to the first process to start and use -iNONE option to avoid creating rsyslog.pid in the first place.